interop/ksp/test-processor/src/main/kotlin/com/squareup/kotlinpoet/ksp/test/processor/TestProcessor.kt

```diff
@@ -66,6 +68,32 @@ class TestProcessor(private val env: SymbolProcessorEnvironment) : SymbolProcess
             .filterNot { it.shortName.getShortName() == "ExampleAnnotation" }
             .map { it.toAnnotationSpec() }.asIterable(),
         )
+        val allSupertypes = decl.superTypes.toList()
+        val superclassReference = if (allSupertypes.isNotEmpty()) {
+          allSupertypes[0].takeIf {
+            val resolved = it.resolve()
+            resolved is KSClassDeclaration && resolved.classKind == ClassKind.CLASS
+          }
+        } else {
+          null
+        }
+        val superInterfaces = if (superclassReference != null) {
+          allSupertypes.drop(1)
```

Comment:
Are we guaranteed to have the superclass at a specific position in the `superTypes` list? Would it be safer to do `filterNot { it == superclassReference }`?

Reply:
Good catch [bf481db](https://github.com/square/kotlinpoet/pull/1529/commits/bf481db9c82a4e2d273aa727fda8dac4a32eb7a4)